kroshakov/implement-remind-repository-and-service-methods

danone05 opened the pull request:
Реализация методов в RemindRepository, RemindService

danone05 committed:
delete all print methods

remove console output and use logging

fix

Refactor: add logging, use getRemindById and List

Заменён вывод через System.out на SLF4J, методы сервиса используют getRemindById,
в репозитории возвращается List вместо ArrayList.

replace int status with RemindStatus enum, add toString(), and update documentation

upd

add-remind-deletion-methods

Methods Reminder

Реализация методов в RemindRepository, RemindService

danone05 committed:
Methods Reminder

Реализация методов в RemindRepository, RemindService

WallKing110 commented:
@danone05 название ветки не совсем говорящее, нужно было бы пополнее написать 
@IronOxidIII и давайте будем название ветки писать в виде фамилия/о_чем_задание

danone05 committed:
add-remind-deletion-methods

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,59 @@
     }
 
-    public ArrayList<Remind> getRemindById(long id) {
-        //TODO
-        return null;
+    /**
+     * Получить все напоминания по userId
+     */
```
Добавь описание аргумента

WallKing110 replied:
@danone05 
Закрой тред сам, плз
```suggestion
    /**
     * Получить одно напоминание по его идентификатору.
     * @param Id напоминания.
     * @return объект {Remind}, если найден; иначе {null}.
     */
```

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,59 @@
+     * Получить все напоминания по userId
+     */
+    public ArrayList<Remind> getRemindsById(long id) {
+        ArrayList<Remind> result = new ArrayList<>();
```
Разве var нельзя использовать?

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,59 @@
+     */
+    public ArrayList<Remind> getRemindsById(long id) {
+        ArrayList<Remind> result = new ArrayList<>();
+        for (Remind r : reminds) {
```
Вместо Remind  var нельзя?

danone05 committed:
upd

WallKing110 commented on .idea/vcs.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
```
@danone05 что это за изменение?

danone05 replied:
@WallKing110 inteliJ сама постоянно это делает при открытии проекта

WallKing110 replied:
@danone05 В гите исключай его из коммита на всякий случай. ну или ладно

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
 
 import java.util.ArrayList;
 
-/**
- * Репозиторий заметок.
- */
```
@danone05 Зачем удалил документацию?

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
-    private ArrayList<Remind> reminds;
+    private final ArrayList<Remind> reminds;
+
+    public ArrayList<Remind> getAll() {
```
@danone05 Допиши комменты

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+    private final ArrayList<Remind> reminds;
+
+    public ArrayList<Remind> getAll() {
+        return new ArrayList<>(reminds);
```
@danone05 Как оноработает? Если я изменю этот возвращенный лист, то и оригинал изменится? Или создадутся копии этих напоминаний?

danone05 replied:
@WallKing110 Возвращается новый список, поэтому если изменить сам список (добавить или удалить элементы),оригинальный reminds не изменится. Но внутри оба списка содержат ссылки на одни и те же объекты Remind, так что изменение полей в самих напоминаниях отразится везде.

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+
+    public ArrayList<Remind> getAll() {
+        return new ArrayList<>(reminds);
+    }
```
@danone05 Перемести публичный метод вниз, за конструктор

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
-        return null;
+    /**
+     * Получить все напоминания по userId.
+     * @param id идентификатор пользователя, для которого нужно найти напоминания
```
@danone05 Точка в конце

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+    /**
+     * Получить все напоминания по userId.
+     * @param id идентификатор пользователя, для которого нужно найти напоминания
+     * @return список напоминаний, принадлежащих указанному пользователю
```
@danone05 Точка в конце

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
-        return null;
+    /**
+     * Получить все напоминания по userId.
+     * @param id идентификатор пользователя, для которого нужно найти напоминания
```
@danone05 Тафтология: id это уникальный ИДЕНТИФИКАТОР, получается уникальный идентификатор индентификатор пользователя

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+    public ArrayList<Remind> getRemindsById(long id) {
+        var result = new ArrayList<Remind>();
+        for (var r : reminds) {
+            if (r.getUserId() == id) result.add(r);
```
@danone05 
```suggestion
            if (r.getUserId() == id) {
                result.add(r);
             }
```

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
 
+    /**
+     * Получить все напоминания по userId.
+     * @param id идентификатор пользователя, для которого нужно найти напоминания
+     * @return список напоминаний, принадлежащих указанному пользователю
+     */
     public ArrayList<Remind> getRemindsByUser(User user) {
-        //TODO
-        return null;
```
@danone05 Я здесь налажал. Давай сделаем так: `getRemindsById` переименуем в `getRemindById` и оно будет возвращать заметку по id заметки. А `getRemindsByUser` все правильно, но будет брать не User-а как аргумент, а userId.   
Кстати, поправь комменты по аналогии с методом выше

danone05 replied:
@WallKing110 Done

WallKing110 replied:
@danone05 давай не сокращать так сильно, remind вместо remind. Закрой тред сам, плз

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
     }
 
+    /**
+     * Добавить напоминание
```
@danone05 точка в конце

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
 
+    /**
+     * Добавить напоминание
+     * @param remind объект напоминания, который нужно добавить;
```
@danone05 
В конце точка.  
Давай перефразируем, просто напоминание. Не обязательно говорить что это объект напоминания

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+        if (remind != null) {
+            reminds.add(remind);
+        }
+    }
```
@danone05 Давай сделаем здесь так:
Если remind = null, то мы выбросим какое-нибудь исключение, типо `IllegalArgumentException`, посмотри в гпт какое лучше выбросить.
И не забудь задокумментировать с помощью `@throws`

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+    }
+
+    /**
+     * Удалить одно напоминание по его ID.
```
@danone05 
```suggestion
     * Удалить напоминание по id.
```

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+
+    /**
+     * Удалить одно напоминание по его ID.
+     * @param remindId идентификатор напоминания, которое нужно удалить
```
@danone05 тафтология, см другой тред
точка в конце

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+     * Удалить одно напоминание по его ID.
+     * @param remindId идентификатор напоминания, которое нужно удалить
+     * @return {true}, если напоминание было найдено и удалено;
+     * {false}, если напоминание с таким ID не найдено
```
@danone05 проверь, что все правильно отображается. Наведи на метод в idea, и посмотри как оно отображается. Предлагаю сделать по-другому. Если не найдено - выбрасываем исключение. И сделать void-ом

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+
+    /**
+     * Удалить все напоминания, принадлежащие пользователю с указанным идентификатором.
+     * @param userId идентификатор пользователя, чьи напоминания нужно удалить
```
@danone05 тафтология и точка

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+        int before = reminds.size();
+        reminds.removeIf(r -> r.getUserId() == userId);
+        return before - reminds.size();
     }
```
@danone05 Зачем возвращать здесь количество удаленных записей? Не вижу варианта, где оно будет использоваться, только если в дебаге. Придется в будущем смотреть а подчеркивания от ide, что занчение нигде не использвется

danone05 replied:
@WallKing110 Done 
Переделал,теперь ничего не выводит

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -5,29 +5,71 @@
+     */
+    public int deleteRemindsByUserId(long userId) {
+        int before = reminds.size();
+        reminds.removeIf(r -> r.getUserId() == userId);
```
@danone05 Оно правильно работает? Типо удаляет все вхождения, а не только одно?

danone05 replied:
@WallKing110 {boolean removeIf(Predicate<? super E> filter)}
Удаляет все элементы, удовлетворяющие условию filter.
По док Java,так что удаляет все.

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,89 @@
+        }
+
+        System.out.println("Список всех напоминаний:");
+        for (Remind remind : reminds) {
```
@danone05 на var нельзя заменить?

danone05 replied:
@WallKing110 Done

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,89 @@
+        ArrayList<Remind> reminds = remindRepository.getAll();
+        for (Remind remind : reminds) {
+            if (remind.getId() == remindId) {
+                remind.setStatus(0); //
```
@danone05 Используй здесь enum со значениями статуса, а не пустое число
Убери, плз, пустой коммент после строки

danone05 replied:
@WallKing110 Done
Немного переделал enum

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,89 @@
+            }
+        }
+        System.out.println("Напоминание с ID " + remindId + " не найдено.");
     }
```
@danone05 Тоже мой косяк(. Давай будем передавать аргументом новый текст напоминания, а не взаимодействовать с пользователем здесь?

danone05 replied:
@WallKing110 Done
Поменял логику

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,89 @@
+        for (Remind remind : reminds) {
+            if (remind.getId() == remindId) {
+                System.out.println("Информация о напоминании:");
+                System.out.println(remind);
```
@danone05 Насколько я знаю, я не переопределял `toString` в `Remind`. Здесь он выводит адрес в памяти, да?

danone05 replied:
@WallKing110 Да, пока показывает адрес объекта. Добавлю toString() в Remind, чтобы выводить id, текст и статус.

danone05 replied:
@WallKing110 Done

danone05 committed:
replace int status with RemindStatus enum, add toString(), and update documentation

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/domain/Remind.java:
OffSetDateTime time тип мне нравится, тут не очень подходящий выбран

LocalTime frequencyOfRepetition - это время суток, а не интервал/продолжительность.

Лучше тут заиспользовать:

private OffsetDateTime time;

private Duration frequencyOfRepetition;

Пример - 
OffsetDateTime time = OffsetDateTime.now();

// Использование:
Duration everyHour = Duration.ofHours(1);
Duration every30Min = Duration.ofMinutes(30);
Duration everyDay = Duration.ofDays(1);
Duration everyWeek = Duration.ofDays(7);

// Вычислить следующее напоминание:
OffsetDateTime nextReminder = time.plus(everyHour);

danone05 replied:
Исправил, заменил LocalTime на Duration для частоты повторения. Теперь используется OffsetDateTime + Duration.

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -8,26 +8,94 @@
-    public ArrayList<Remind> getRemindById(long id) {
-        //TODO
-        return null;
+    public ArrayList<Remind> getAll() {
```
Давайте возвращать тип List<> и не привязывать метод к конкретной реализации

Внутри уже можете создавать эрейлист и возвращать его

danone05 replied:
Исправил, теперь метод возвращает List, реализация внутри осталась ArrayList.

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -8,26 +8,94 @@
+     * @return список напоминаний, принадлежащих этому пользователю.
+     */
+
+    public ArrayList<Remind> getRemindsByUser(long userId) {
```
Давайте возвращать тип List<> и не привязывать метод к конкретной реализации

danone05 replied:
Исправил

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,91 @@
+                remind.setStatus(RemindStatus.Deactivated);
+            }
+        }
+                System.out.println("Напоминание ID " + remindId + " отключено.");
```
А давайте вместо принта писать логи правильно)
Будем использовать самую популярную либу - Slf4j

// Добавить в pom.xml: (найдите актуальную версию)
<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-api</artifactId>
    <version>2.0.9</version>
</dependency>
<dependency>
    <groupId>ch.qos.logback</groupId>
    <artifactId>logback-classic</artifactId>
    <version>1.4.11</version>
</dependency>


// В коде:
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

 log.debug("Adding remind: {}", remind.getId());

 log.info("Remind added successfully: {}", remind.getId());

log.error("Failed to add remind", e);


Для настройки вывода логов ( в файл или в консоль) нужно создать logback.xml
Прочитайте про это

danone05 replied:
Добавил SLF4J + Logback, заменил System.out.println() на логирование

WallKing110 replied:
@danone05 отписал в ревью по изменению. Или сохрани функционал старый + добавь логирование, или поменяй название методов и описание. Лучше первый вариант
@lostflydev Что думаешь?

lostflydev replied:
@WallKing110 Валидное замечание. Лучше первый вариант. Тут возможно друг друга недопоняли, где нужно логирование, а где вывод в консоль, не стоило все принты менять на логи

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,91 @@
+    for (var remind : reminds) {
+        if (remind.getId() == remindId) {
+            remind.setText(newText);
+            System.out.println("Текст напоминания успешно изменён.");
```
логи

danone05 replied:
✅

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,91 @@
+ * @throws IllegalArgumentException если напоминание с таким ID не найдено.
+ */
+public void changeRemindText(long remindId, String newText) {
+    var reminds = remindRepository.getAll();
```
Давайте сделаем в репозитории метод getRemindById(), инкапсулируем половину логики отсюда в репозитории

danone05 replied:
Метод getRemindById() уже был в RemindRepository, сервис теперь использует его вместо getAll()

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,91 @@
     public void printRemindInfo(long remindId) {
-        //TODO
-        return;
+        ArrayList<Remind> reminds = remindRepository.getAll();
```
Также пригодится метод getRemindById, иначе сейчас есть дублирование кода тут и в изменении текста

danone05 replied:
Использую getRemindById в printRemindInfo

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,91 @@
+    public void deleteRemind(long remindId) {
+        boolean removed = remindRepository.deleteRemindById(remindId);
+        if (removed) {
+            System.out.println("Напоминание с ID " + remindId + " удалено.");
```
логи

danone05 replied:
✅

danone05 committed:
Refactor: add logging, use getRemindById and List

Заменён вывод через System.out на SLF4J, методы сервиса используют getRemindById,
в репозитории возвращается List вместо ArrayList.

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,102 @@
+     * Список всех напоминаний, хранящихся в репозитории.
+     * Используется как внутренняя коллекция для хранения данных в памяти.
+     */
+
```
@danone05 лишняя строка

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,102 @@
+
+    private final List<Remind> reminds;
+
+    /**
+     * Возвращает копию списка всех напоминаний, сохранённых в репозитории.
+     *
+     * @return новый список, содержащий все текущие напоминания.
+     */
```
@danone05 Слетела дока, переставь на место, плз

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,102 @@
+     * @param ID пользователя.
+     * @return список напоминаний, принадлежащих этому пользователю.
+     */
+
```
@danone05 лишняя строка

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,102 @@
+     * Добавляет новое напоминание в репозиторий.
+     * @param remind напоминание, которое нужно добавить.
+     * @throws IllegalArgumentException если remind равен {null}.
+     */
```
@danone05 выбери что-то одно

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,88 @@
+ * Сервис для работы с напоминаниями.
+ */
 public class RemindService {
-    private RemindRepository remindRepository;
+    private final RemindRepository remindRepository;
+    private static final Logger log = LoggerFactory.getLogger(RemindService.class);
```
@danone05 раздели их пустой строкой

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,88 @@
+        for (var remind : reminds) {
+            log.debug("{}", remind);
+        }
     }
```
@danone05 Я так понимаю ты исправил по ревью, но теперь эта функция не работает как надо, нужно же в консоль выводить, а в лог наверное давай дублировать то, что выводится в консоль. Или измени описание метода (вместе с названием)

danone05 replied:
@WallKing110 вернул вывод ,в лог дублируется

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,88 @@
-        //TODO
-        return;
+        var remind = remindRepository.getRemindById(remindId);
+        if (remind == null) {
```
@danone05 раздели наверное их пустой строкой по всему мр-у, читаемость лучше будет. Ну или если не хочешь, то просто закрой тред

danone05 replied:
@WallKing110 ✅

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,88 @@
+        }
+
+        log.info("Информация о напоминании: {}", remind);
     }
```
@danone05 Здесь аналогично, метод должен выводить в консоль, а по факту логирует

danone05 replied:
@WallKing110 сделал вывод и лог

danone05 committed:
fix

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
-        return;
+        var reminds = remindRepository.getAll();
+        if (reminds.isEmpty()) {
+            System.out.println("Нет ни одного напоминания.");
```
зачем и лог и принт? давай везде где информация не для вывода в консоль юзеру, а служебная, поубираем принты

lostflydev replied:
Зачем и принт и лог? Давай в местах, где служебные выводы, оставлять токо логи, во всех местах поправь плиз

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
+        }
+
+        System.out.println("Список всех напоминаний:");
+        log.info("Список всех напоминаний:");
```
аналогично

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
+        for (var remind : reminds) {
+            if (remind.getId() == remindId) {
+                remind.setStatus(RemindStatus.Deactivated);
+                log.info("Напоминание с ID {} отключено.", remindId);
```
используй s-строки, и сможешь использовать ID {remind.getId()}

lostflydev replied:
Давай использовать s строки и не использовать конкатенацию в таких местах

s"Напоминание с ID ${remind.getId()} отключено."

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
+                return;
+            }
+        }
+        log.warn("Напоминание с ID {} не найдено.", remindId);
```
s строки

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
+
+    if (remind == null) {
+        log.warn("Напоминание с ID {} не найдено.", remindId);
+        throw new IllegalArgumentException("Напоминание с ID " + remindId + " не найдено.");
```
красивее s строки

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/service/RemindService.java:
```diff
@@ -1,27 +1,100 @@
+        
+        if (remind == null) {
+            System.out.println("Напоминание с ID " + remindId + " не найдено.");
+            log.warn("Напоминание с ID {} не найдено.", remindId);
```
два вывода, оставим лог, дальше не буду комментить, сделайте для всех мест

lostflydev replied:
два вывода, оставим лог, дальше не буду комментить, сделайте для всех мест

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,91 @@
+    public void deleteRemindById(long remindId) {
+        boolean removed = reminds.removeIf(r -> r.getId() == remindId);
+        if (!removed) {
+            throw new IllegalArgumentException("Remind with id " + remindId + " not found");
```
Давай использовать s строки и не использовать конкатенацию в таких местах

lostflydev commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,91 @@
+    public void deleteRemindsByUserId(long userId) {
+        boolean removed = reminds.removeIf(r -> r.getUserId() == userId);
+        if (!removed) {
+            throw new IllegalArgumentException("No reminds found for userId " + userId);
```
Давай использовать s строки и не использовать конкатенацию в таких местах

danone05 committed:
remove console output and use logging

lostflydev reviewed (approved):
От меня ок

danone05 committed:
delete all print methods

danone05 committed:
Merge branch 'Remind' of https://github.com/IronOxidIII/TRB_TheReminderBot into Kroshakov/implement-remind-repository-and-service-methods

WallKing110 commented on thereminderbot/src/main/java/org/thereminderbot/repository/RemindRepository.java:
```diff
@@ -3,31 +3,91 @@
+    /**
+     * Получить все напоминания, принадлежащие пользователю с указанным userId.
+     */
+
```
@danone05 до сих пор лишняя строка
